### ISSUE TITLE
Replacement unicode character in text extraction

### ISSUE THREAD

**Issue by JorjMcKie** [pymupdf/PyMuPDF#2954]
### Discussed in https://github.com/pymupdf/PyMuPDF/discussions/2952

<div type='discussions-op-text'>

<sup>Originally posted by **indigoviolet** January  1, 2024</sup>
[test.pdf](https://github.com/pymupdf/PyMuPDF/files/13806565/test.pdf)

```shell
❯ mutool draw -F text test.pdf | head    
����������������� ������ ����
NYPA2514    12/06/21

�������������������
���������������������������������

��
����������������

������
```

And of course, PyMuPDF also has 65333s

```shell
❯ pdftotext test.pdf - | head
NYPA2514

Partner's share of
20

modifications (see instructions)

State additions
A ' Total amount
```

pdfplumber also parses this file successfully. 

Any suggestions?</div>

**Comment by JorjMcKie** [comment]
This is the reference to the MuPDF issue: https://bugs.ghostscript.com/show_bug.cgi?id=707440

**Comment by JorjMcKie** [comment]
This turns out to an interesting one!
Our friends from MuPDF have looked at the case:
The PDF is broken and defines an erroneous `/ToUnicode` table. One **_could_** look at the glyph numbers (currently not available in standard text extraction) and simply **_assume_** them to _**be**_ the Unicode value.
This is obviously what other tools do.
MuPDF is evaluating options for a solution.

**Comment by JorjMcKie** [comment]
MuPDF has developed a solution which fixes the problem!

**Comment by indigoviolet** [comment]
> MuPDF has developed a solution which fixes the problem!

Great, thank you. Can you help me understand the timeline for such fixes to roll out through MuPDF and then PyMuPDF?

**Comment by JorjMcKie** [comment]
This change will probably be part of MuPDF version 1.24. I have no schedule for its publication yet.
But please feel free to voice your interest in [this](https://discord.com/channels/770681584617652264/770685907279282196) public MuPDF Discord channel.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.9.
